Dockerfile

```
@@ -1,49 +1,30 @@
-FROM nvidia/cuda:9.2-cudnn7-devel-ubuntu16.04
-
-# create user
-ARG user_id
-ARG user_name
-ARG group_id
-ARG group_name
-RUN groupadd -g ${group_id} ${group_name} && \
-    useradd -u ${user_id} -g ${group_name} -s /bin/bash -m ${user_name} && \
-    echo "${user_name} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers && \
-    chown -R ${user_name}:${group_name} /home/${user_name}
-
-# Default
+FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu16.04
+ARG PYTHON_VERSION=3.7
+ARG WITH_TORCHVISION=1
 RUN apt-get update && apt-get install -y --no-install-recommends \
          build-essential \
          cmake \
          git \
          curl \
-         wget \
-         vim \
          ca-certificates \
          libjpeg-dev \
-         libgl1-mesa-dev \
-         libpng-dev \
-         build-essential \
-         zip \
-         unzip \
-         libpng-dev &&\
-    rm -rf /var/lib/apt/lists/*
+         libpng-dev && \
+     rm -rf /var/lib/apt/lists/*
 
-# Python Anaconda default.
-RUN wget -q https://repo.anaconda.com/archive/Anaconda3-5.3.1-Linux-x86_64.sh -O ~/anaconda.sh && \
-    /bin/bash ~/anaconda.sh -b -p /opt/conda && \
-    rm ~/anaconda.sh
-# Install PyTorch V1.
-ENV PATH /opt/conda/bin:$PATH
-ARG PYTHON_VERSION
-RUN conda install -y python=$PYTHON_VERSION && \
-    conda install -y -c conda-forge feather-format && \
-    conda install -y -c conda-forge jupyterlab && \
-    conda install -y -c conda-forge jupyter_contrib_nbextensions && \
-    jupyter contrib nbextension install --system
-RUN conda install -y pytorch torchvision -c pytorch
-RUN conda install -y -c conda-forge tensorflow && \
-    conda clean -y --all && \
-    pip install --no-cache-dir tensorboardX
 
+RUN curl -o ~/miniconda.sh -O  https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
+     chmod +x ~/miniconda.sh && \
+     ~/miniconda.sh -b -p /opt/conda && \
+     rm ~/miniconda.sh && \
+     /opt/conda/bin/conda install -y python=$PYTHON_VERSION numpy pyyaml scipy ipython mkl mkl-include ninja cython typing && \
+     /opt/conda/bin/conda install -y -c pytorch magma-cuda101 && \
+     /opt/conda/bin/conda install -y -c pytorch pytorch torchvision && \
+     /opt/conda/bin/conda install -y -c anaconda pytest && \
+     /opt/conda/bin/conda clean -ya
 ENV PATH /opt/conda/bin:$PATH
-WORKDIR /src
+
+RUN /opt/conda/bin/conda install -y -c conda-forge tensorflow && \
+    /opt/conda/bin/conda clean -y --all
+
```

with the reference of [Adding entrypoint](https://circleci.com/docs/ja/2.0/custom-images/#%E3%82%A8%E3%83%B3%E3%83%88%E3%83%AA%E3%83%9D%E3%82%A4%E3%83%B3%E3%83%88%E3%81%AE%E8%BF%BD%E5%8A%A0)

```suggestion

LABEL com.circleci.preserve-entrypoint=true
```